LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/iframe-zoom-nested.html

```diff
@@ -26,24 +26,24 @@
 </head>
 
 <body>
-
-<iframe id="baseline" src="resources/nested-iframe.html" scrolling="no"></iframe>
-<iframe id="zoom-child" src="resources/nested-iframe.html?zoom=2" scrolling="no"></iframe>
-<iframe id="zoom-top" class="zoom" src="resources/nested-iframe.html" scrolling="no"></iframe>
-<iframe id="zoom-top-child" class="zoom" src="resources/nested-iframe.html?zoom=2" scrolling="no"></iframe>
-
+    
+    <iframe id="baseline" src="resources/nested-iframe.html" scrolling="no"></iframe>
+    <iframe id="zoom-child" src="resources/nested-iframe.html?zoom=2" scrolling="no"></iframe>
+    <iframe id="zoom-top" class="zoom" src="resources/nested-iframe.html" scrolling="no"></iframe>
+    <iframe id="zoom-top-child" class="zoom" src="resources/nested-iframe.html?zoom=2" scrolling="no"></iframe>
+    
 <script>
-// Wait for all iframes to load before allowing the screenshot
-let loadedCount = 0;
-const iframes = document.querySelectorAll('iframe');
-const totalFrames = iframes.length;
-
-iframes.forEach(iframe => {
-  iframe.onload = function() {
-    loadedCount++;
-    if (loadedCount === totalFrames) {
-      document.documentElement.classList.remove("reftest-wait");
-    }
-  };
-});
-</script>
+    let finishTest = () => {
+        document.documentElement.classList.remove("reftest-wait");
+    };
+    let loadedCount = 0;
+    const waitFor = 4;
+    addEventListener("load", () => {
+        addEventListener("message", ({ data }) => {
```

Comment:
Could we add a comment to explain what's being done here?

`// We need to wait for the nested iframe (leaf.html) to load, in addition to the top-level iframe (nested-iframe.html).`